testing/src/scenario/state.py

```diff
@@ -1502,6 +1534,24 @@ def __post_init__(self):
             if not isinstance(val, frozenset):
                 object.__setattr__(self, name, frozenset(val))
 
+    def __deepcopy__(self, memo: dict[int, Any] | None = None) -> State:
```

Comment:
Can you add a comment explaining why this class needs a `__deepcopy__` (when others don't)?